### PR TITLE
feat(wizard): added possibility to ignore dialog dependency

### PR DESCRIPTION
**Pull request by rubenbarroshv**

### CODE REVIEW

**Review by andrew-cockayne — COMMENTED** [review]
In the "Fullscreen Wizard" story the header and footer are much higher than they need to be, and the styling is wrong. Ideally the "fullscreen" mode would look identical to the "dialog" version.

That said, I see how the approach is working to solve the issue, but I do not think the approach is correct. For example, a table does not have any considerations for its container. It can be in a screen, a dialog, a drawer or any other component. What makes a Wizard any different to a table? Why is it not simple a component which can be applied to any parent?

Changing the Wizard component in the way described above would remove all "display" related props, `open`, `onClone`, etc. It would also mean it can be styled independently of the dialog. I appreciate doing so may be more work (from where it is now), but equally, I expect if left as is we may face further issues with how or where the Wizard is displayed.

**Comment by rubenbarroshv** [comment]
> In the "Fullscreen Wizard" story the header and footer are much higher than they need to be, and the styling is wrong. Ideally the "fullscreen" mode would look identical to the "dialog" version.
> 
> That said, I see how the approach is working to solve the issue, but I do not think the approach is correct. For example, a table does not have any considerations for its container. It can be in a screen, a dialog, a drawer or any other component. What makes a Wizard any different to a table? Why is it not simple a component which can be applied to any parent?
> 
> Changing the Wizard component in the way described above would remove all "display" related props, `open`, `onClone`, etc. It would also mean it can be styled independently of the dialog. I appreciate doing so may be more work (from where it is now), but equally, I expect if left as is we may face further issues with how or where the Wizard is displayed.

Both the Header and Footer look bigger because I styled them that way on the story component. Just to show how the user could style the components.

The "fullscreen" might be a deceptive word/flag to use, it's just a plain "div", you could put this component anywhere you would see fit, and then style it to your liking.

I wasn't thinking of making those props optional, since I am not using those on the "fullscreen" version.

@andrew-cockayne what do you think?

File: packages/lab/src/Wizard/stories/Wizard.stories.js [hunk 1]
```diff
@@ -348,3 +348,93 @@ export const ComponentBreakDown = () => {
...
+    flex: 1,
+    overflowY: "auto",
+  },
+  customActions: {
+    backgroundColor: "gray",
+    height: 200,
+  },
+}));
+
+export const FullscreenWizard = () => {
```
[the remark this answers is not in the document]
Comment (rubenbarroshv) [reply packages/lab/src/Wizard/stories/Wizard.stories.js:380]: my spell checker accepts Fullscreen as a full word, but I can change this 👍

File: packages/lab/src/Wizard/WizardContainer/WizardContainer.js [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
       classes={{ closeButton: classes.closeButton, paper: classes.paper }} // overrides css
-      className={clsx(className, classes.root)}
+      className={className}
       open={open}
       onClose={handleClose}
       {...others}
     >
-      {children}
-    </HvDialog>
+      {React.Children.map(arrayChildren, (child) => React.cloneElement(child, { fullscreen }))}
```
[the remark this answers is not in the document]
Comment (rubenbarroshv) [reply packages/lab/src/Wizard/WizardContainer/WizardContainer.js]: Yes, you are right.

File: packages/lab/src/Wizard/stories/Wizard.stories.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrew-cockayne) [reply packages/lab/src/Wizard/stories/Wizard.stories.js]: Yes, nuance with English. In this context I believe it should be FullScreenWizard as a component name is typically a noun. I'd expect there would be other opionons, but that's my preference.

**Pull request closed without merging**